### ISSUE TITLE
Playing stops, can neither restart nor switch to other song

### ISSUE THREAD

**Issue by chetmurthy** [LMS-Community/slimserver#1013]
I'm using:
Server: Logitech Media Server Version: 8.4.1 - 1709721691 on Ubuntu Linux 22.04.4 LTS
Players: squeezeslave on Max2Play for Raspberry Pi 3

And for quite a while (I think, since version 8.4.1),  when I start playing music randomly, eventually a song will finish, but music will not advance to the next song, I cannot
1. back up
2. change place in song (e.g. by clicking on the progress bar)
3. restart the song
4. choose a different song.  If I click "Random songs" I'll get a different song, but it won't play -- I'll get silence.

I can get play to restart only by doing a "stop/restart" at the level of the Unix prompt.  The songs that it hangs on aren't special, and *after restarting* I can play them by selecting them and there's no problem.

1. This has been going on for a good while
7. I forget when it started, but it was immediately after an upgrade
8. at the time, I hoped that I could just update to a nightly and it'd go away, and then, well, the workaround works, so I neglected to file an issue/report.

I don't know exactly what sort of diagnostic info I should capture, but I'm an experienced UNIX jock, happy to do whatever is needed to get the info.

**Comment by michaelherger** [comment]
You say "when I start playing music randomly": this issue is not limited to playing _randomly_, but it would happen with any song randomly?

Please check your server.log (see Settings/Information).

And then I'd suggest you join https://forums.slimdevices.com where there are a ton of helpful an knowledgeable people. Many more than here.

**Comment by chetmurthy** [comment]
Sorry, what I meant was, I always play music with "Random Mix > Random songs".  Music will play, and after some time (a few hours, typically) a song will be playing, and after it finishes the players go silent. I cannot forward past, nor back up, nor switch to a different song, nor again choose a random playlist as above.  The only thing that works, is to stop/start the entire server apparatus.

I just downgraded to 8.4.0, and if that doesn't freeze up, I'll then upgrade back up, and see if I find anything in the server log.

**Comment by michaelherger** [comment]
Always check server.log for any hints. Your server wouldn't go into standby mode, would it?

**Comment by chetmurthy** [comment]
If by "standby mode" you're referring to the server machine/operating
system, no, it's functioning.  Typically, I keep a login shell on that
machine, and when the problem happens,

1. I still have that shell and can run commands
2. I can perform "Random Mix > Random songs" and it actually takes effect
in the Web UI
3. but the music doesn't change (typically this is after the music has
stopped, so it doesn't start again)

P.S. I should add that this is my home server, so it's never on standby: it's always awake and running.

**Comment by chetmurthy** [comment]
OK, I downgraded to 8.4.0, and the problem still occurs.  Must have been an earlier version where the problem didn't happen (I'll figure that out).  Meanwhile, here's what was at the bottom of server.log (when playback stopped, I tried clicking "next song" (right-arrow) a few times):

[24-03-17 11:03:45.0308] Slim::Player::StreamingController::_BadState (317) b8:27:eb:e0:cd:b0: event Skip received while in invalid state BUFFERING-TRACKWAIT
[24-03-17 11:03:45.0311] Slim::Player::StreamingController::_eventAction (287) b8:27:eb:e0:cd:b0:  Skip with action Slim::Player::StreamingController::_BadState resulted in invalid state BUFFERING-TRACKWAIT
[24-03-17 11:03:47.0084] Slim::Player::StreamingController::_BadState (317) b8:27:eb:e0:cd:b0: event StatusHeartbeat received while in invalid state BUFFERING-TRACKWAIT
[24-03-17 11:03:47.0087] Slim::Player::StreamingController::_eventAction (287) b8:27:eb:e0:cd:b0:  StatusHeartbeat with action Slim::Player::StreamingController::_BadState resulted in invalid state BUFFERING-TRACKWAIT
[24-03-17 11:03:52.0093] Slim::Player::StreamingController::_BadState (317) b8:27:eb:e0:cd:b0: event StatusHeartbeat received while in invalid state BUFFERING-TRACKWAIT
[24-03-17 11:03:52.0096] Slim::Player::StreamingController::_eventAction (287) b8:27:eb:e0:cd:b0:  StatusHeartbeat with action Slim::Player::StreamingController::_BadState resulted in invalid state BUFFERING-TRACKWAIT
[24-03-17 11:03:57.0076] Slim::Player::StreamingController::_BadState (317) b8:27:eb:e0:cd:b0: event StatusHeartbeat received while in invalid state BUFFERING-TRACKWAIT
[24-03-17 11:03:57.0079] Slim::Player::StreamingController::_eventAction (287) b8:27:eb:e0:cd:b0:  StatusHeartbeat with action Slim::Player::StreamingController::_BadState resulted in invalid state BUFFERING-TRACKWAIT

**Comment by chetmurthy** [comment]
I upgraded to 8.5.0, and the problem persisted.  Ditto 8.4.0.  Downgrading to 8.3.1 seems to have fixed the problem.  Is there an archive someplace of the nightly amd64 DEBs ?  If so, I could bisect in that collection to find out the earliest nightly release that exhibits the problem?  That might help narrow down the cause ?

**Comment by michaelherger** [comment]
It's easier to compare on Git than from two packages:

https://github.com/LMS-Community/slimserver/compare/public/8.3...public/8.5

**Comment by chetmurthy** [comment]
Yes, my plan was to find two consecutive nightly builds where the first doesn't show the problem, and the second shows it.  Then go to GIT for those builds and compare.  So far fewer changes.  There are several problems with just going straight to GIT:

1. between 8.3.1 and 8.4.0 there will be a bunch of changes, right?
2. If I try to bisect the GIT commit-history between those two releases, I won't know whether each commit yields an ostensibly runnable corpus.  So I could end up going down a lot of ratholes.  At least with nighty releases, somebody else decided that it was safe to release.
3. I don't know the code, so it'll be difficult to figure out what might be problematic
4. I've only ever run LMS as a user, from installed DEB; I'm an experienced systems jock and I could figure it out, but it would take a lot of work to learn how to run LMS from sources.

But the real problem is that I don't know how to build and run from source.  Am hoping if I can install and run nightly DEBs, that won't be necessary.

**Comment by michaelherger** [comment]
Don't worry: nightlies are built automatically, no check involved 😁. 

I'd recommend you take the issue to https://forums.slimdevices.com where there are not only more people willing to help, but also more devs who know what might have changed. And please include some more information about the files you're playing what format? Are you really using squeezeslave rather than squeezelite on m2p?
